contracts/fio.address/fio.address.cpp

```diff
@@ -144,6 +150,51 @@ namespace fioio {
 
         }
 
+        // FIP-39 begin
+        inline void updhandleinf(const string &datavalue, const string &datadesc, const uint64_t &fionameid, const name &actor) {
+            auto handleinfobynameid = handleinfo.get_index<"byfionameid"_n>();
+            auto handleinfo_iter = handleinfobynameid.find(fionameid);
+            if(handleinfo_iter == handleinfobynameid.end()){
```

Comment:
Let's discuss this code section. Adding some notes so I don't forget:

If I am reading this code correctly it looks like:
- Check to see if the fioname has an entry in the table.
  - If not, emplace the new datavalue/datadesc
  - If yes
        - use the item found as the initial value (line 166)
        - go through the table to find if there is a entry with that datadesc already
              - If yes, update that entry
              - If no, it updates the value from line 166

Won't that overwrite the initially found item, even if it is not the same datadesc?

Need to test: 
- Add entry with datavalue1/datadesc1
- Add entry with datavalue2/datadesc2
- Confirm both entries still exist

Reply:
Let's do a comprehensive code walk through offline at your next opportunity.

Reply:
Discussed with Ed. I was reading the string check wrong. Also decided that we are only testing with the assumption there is a single datadesc1 type. Will not test the case above.